src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -0,0 +1,118 @@
+    // React component state. Then I can cross-reference this when adding the child nodes instead of
+    // making a fetch call - which seems impossible to wait for the result of (maybe fetch is the wrong thing to use?)
+    @WebMethod(name = "steps")
+    public void getSteps(StaplerRequest req, StaplerResponse rsp) throws IOException {
+        LOGGER.log(Level.FINE, "PipelineConsoleViewAction getSteps called.");
+        String nodeId = req.getParameter("nodeId");
+        if (nodeId != null) {
+            LOGGER.log(Level.FINE, "PipelineConsoleViewAction getSteps passed nodeId '" + nodeId + "'.");
+            PipelineStepApi stepApi = new PipelineStepApi(target, nodeId);
+            ObjectMapper mapper = new ObjectMapper();
+            LOGGER.log(Level.FINE, "Steps: '" + mapper.writeValueAsString(stepApi.getSteps()) + "'.");
+            rsp.getWriter().append(mapper.writeValueAsString(stepApi.getSteps()));
+        } else {
+            LOGGER.log(Level.FINE, "PipelineConsoleViewAction getSteps not passed nodeId.");
+            rsp.getWriter().append("Error getting console text");
+        }
+    }
+
+    @WebMethod(name = "consoleOutput")
+    public void getConsoleOutput(StaplerRequest req, StaplerResponse rsp) throws IOException {
+        LOGGER.log(Level.FINE, "PipelineConsoleViewAction getConsoleOutput called.");
+        String nodeId = req.getParameter("nodeId");
+        if (nodeId != null) {
+            LOGGER.log(Level.FINE, "PipelineConsoleViewAction getConsoleOutput passed nodeId.");
+            String nodeConsoleText = getLogForNode(nodeId);
+            if (nodeConsoleText != null) {
+                rsp.getWriter().append(nodeConsoleText);
+            } else {
+                rsp.getWriter().append("No console output for node: ").append(nodeId);
+            }
+        } else {
+            LOGGER.log(Level.FINE, "PipelineConsoleViewAction getConsoleOutput not passed nodeId.");
+            rsp.getWriter().append("Error getting console text");
+        }
+    }
+
+    private String getLogForNode(String nodeId) throws IOException {
+        FlowExecution execution = target.getExecution();
+        if (execution != null) {
+            LOGGER.log(Level.FINE, "PipelineConsoleViewAction getConsoleOutput found execution.");
+            FlowNode node = execution.getNode(nodeId);
+            if (node != null) {
+                LOGGER.log(Level.FINE, "PipelineConsoleViewAction getConsoleOutput found node.");
+                LogAction log = node.getAction(LogAction.class);
+                if (log != null) {
+                    ByteArrayOutputStream oututStream = new ByteArrayOutputStream();
+                    log.getLogText().writeLogTo(0, oututStream);
+                    String consoleText = oututStream.toString().trim();
+                    int maxLength = 120;
+                    if (consoleText.length() > maxLength) {
```

Comment:
why is this being limited?
is this long lines?
does blue ocean limit this?

Reply:
This log message to to show that we found the correct console output for the step. I limit it to reduce the noise we add to the logs (if it's a 10MB log for a step, I doubt we want 10MB in the Jenkins logs).
I can also remove this logging if you don't think we'll need it?

Reply:
why wouldn't we want it? doesn't the user need to be able to see all the log?

is it cut off elsewhere?

Reply:
The log will still appear in the console view, just not in the Jenkins master log itself.

Reply:
oh right, I would just remove the loggingn then